QVRWeekView/Classes/Views/WeekView.swift

```diff
@@ -44,6 +44,21 @@ open class WeekView: UIView {
         let firstActiveDay = self.dayScrollView.activeDay
         return firstActiveDay...(firstActiveDay + Int(LayoutVariables.visibleDays - 1))
     }
+    
+    //A date range containing the current days visible on screen
+    public var visibleDateRange: ClosedRange<Date> {
+        let firstActiveDay = self.dayScrollView.activeDay
+        let lastActiveDay = (firstActiveDay + Int(LayoutVariables.visibleDays - 1))
+        return firstActiveDay.getDateWithTime(hours: 0, minutes: 0, seconds: 0)...lastActiveDay.getDateWithTime(hours: 23, minutes: 59, seconds: 59)
+    }
+    
+    //Reloads the events from the delegate
+    public func notifyDataSetChanged() {
+        if let delegate = self.delegate {
+            let range = visibleDateRange
+            delegate.eventLoadRequest(in: self, between: range.lowerBound, and: range.upperBound)
+        }
```

Comment:
Could this entire function maybe be shortened to:
```
delegate?.eventLoadRequest(in: self, 
    between: visibleDateRange.lowerBound,
    and: visibleDateRange.upperBound)
```

Reply:
Yes, though it would be at the minor cost of additional computation as it would calculate the visible date range twice.  Would you prefer it that way?  I'm fine with changing it

Reply:
It is done!